core/src/main/java/org/infinispan/factories/impl/MBeanMetadata.java

```diff
@@ -161,6 +176,27 @@ public Consumer<?> setter(Object instance) {
          return (v) -> ((BiConsumer<Object, Object>) setterFunction).accept(instance, v);
       }
 
+      public Consumer<CounterTracker> counter(Object instance) {
```

Comment:
The usage of all three methods that return a `Consumer` is always:

. Get Consumer
. `assert != null`
. `Consumer#accept`

Wouldn't it be simpler to just add that logic here and take the `CounterTracker` etc as a parameter?

Reply:
done! thanks for the suggestion